### DISCUSSION TITLE
Tree and data must be overridden in order to update the chart

### DISCUSSION THREAD

**Discussion by stockiNail** [kurkle/chartjs-chart-treemap#171] [Ideas]
Going to the last version, I have discovered that if you change the values of the tree (or data) without instantiating new object (or array), the chart will not be updated.

This is because the tree is stored in the controller and checked with new one at every update:

https://github.com/kurkle/chartjs-chart-treemap/blob/f6a7af1e15e21adafcb9b8ba502cf6f3acd3988f/src/controller.js#L112

Maybe it's implemented as designed. But maybe it could be better to check not the instance but the content of the tree.

**Comment by kurkle** [comment]
If I remember correctly, `arrayNotEqual` did not work previously, so this check was always true.
That is how it was designed. If you share the use case, maybe we can figure out if we can improve things.

**Reply by stockiNail** [reply]
I don't have a real use case but in my showcase, I'm allowing the user to "randomize" the data and I didn't create new object but only change the existing one. 
I think the check is not related to `arrayNotEqual` (because a tree could be also an object) but is related to `this._prevTree !== tree`.
But if it is as designed, no need to change it, if we will not be a specific case. It sounds to me a deviation of the "normal" way how CHART.JS can update the chart.

**Reply by kurkle** [reply]
Sounds like the desing is bad. It never worked as designed in previous versions, so it could also be considered a bug :)

**Reply by stockiNail** [reply]
oki! let me copy the discussion to an issue, for tracking
